### PR TITLE
an improved and simplified version of Shutter::App::Directories

### PR DESCRIPTION
**Pull request by ruzhnikov**

### CODE REVIEW

File: share/shutter/resources/modules/Shutter/App/Directories.pm [hunk 1]
```diff
@@ -22,62 +23,56 @@
...
+    SHUTTER_DIR        => "shutter",
+    UNSAVED_DIR        => "unsaved",
+    TEMP_DIR           => "temp",
+    AUTOSTART_DIR      => "autostart",
+    HIDDEN_SHUTTER_DIR => ".shutter",
+    PROFILES_DIR       => "profiles"
+};
 
-	my $self = {};
+sub create_if_not_exists : prototype($) {
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/App/Directories.pm]: I thought prototypes aren't recommended generally, unless you're trying to mimic some perl builtins

**Comment by Photon89** [comment]
Thanks for the recent work on Shutter! I wonder if these changes fix this very old issue https://github.com/shutter-project/shutter/issues/67?

File: share/shutter/resources/modules/Shutter/App/Directories.pm (hunk 1 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/Directories.pm]: Usually yes, it's not worth to use them in the code, especially in OOP. But for small, often used functions it's OK. The main advantage of using prototypes is checking their invoking in compile time. If list of arguments in caller mismatches with prototype of the function, you will get an error and code won't run at all. But it works only with using `use` construction, not `require` and hence, now it isn't actual.
So, until `use Shutter::App::Directories` isn't used in the project, I agree that this is not needed here. I'll remove it

**Comment by ruzhnikov** [comment]
> Thanks for the recent work on Shutter! I wonder if these changes fix this very old issue #67?

I'm not sure about it. I just rewrote the code of Shutter::App::Directories in functional style but I didn't change the work logic of functions themselves. I'll check the #67 and add fixes if it's needed.

File: share/shutter/resources/modules/Shutter/App/Directories.pm (hunk 1 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/Directories.pm]: ok, it's done in a6516a89

**Comment by ruzhnikov** [comment]
@Photon89 I've checked the issue #67 and no, this PR doesn't fix it. I think that discussion may be continued in that issue.

**Comment by Photon89** [comment]
Sure! Thanks for having a look at it! The bug has been imported from Launchpad with high priority, but I think, it's not that pressing as it exists since 2012 already. :smile:

**Pull request merged**